api/src/main/java/org/openmrs/module/providermanagement/ProviderManagementUtils.java

```diff
@@ -99,6 +104,66 @@ public static Date clearTimeComponent(Date date) {
+            throws PersonIsNotProviderException {
+
+        ProviderManagementService providerManagementService = Context.getService(ProviderManagementService.class);
+        List<ProviderPersonRelationship> supervisors = new ArrayList<ProviderPersonRelationship>();
+        Person person = provider.getPerson();
+        List<Person> supervisorPersons = providerManagementService.getSupervisorsForProvider(person);
+        for (Person supervisor : supervisorPersons) {
+            List<Provider> providersByPerson = providerManagementService.getProvidersByPerson(supervisor, true);
+            if (providersByPerson !=null && providersByPerson.size() > 0) {
+                RelationshipType supervisorRelationshipType = providerManagementService.getSupervisorRelationshipType();
+                Relationship supervisorRelationship = null;
+                Provider supervisorProvider = providersByPerson.get(0);
+                List<Relationship> relationships = Context.getPersonService().getRelationships(supervisor,
+                        person, supervisorRelationshipType, null);
+                if (relationships != null && relationships.size() > 0 ){
+                    for (Relationship relationship : relationships) {
+                        if ( ( supervisorRelationship == null && relationship.getEndDate() == null ) ||
+                                (supervisorRelationship != null && relationship.getEndDate() == null
+                                        && relationship.getStartDate().after(supervisorRelationship.getStartDate()))) {
+                            // select the most active relationship
+                            supervisorRelationship = relationship;
+                        }
+                    }
+                }
+                supervisors.add(new ProviderPersonRelationship(
+                        supervisorProvider.getPerson(),
+                        supervisorProvider.getIdentifier(),
+                        supervisor.getId(),
+                        supervisorRelationship,
+                        supervisorRelationshipType));
+            }
+        }
+        return supervisors;
+    }
+
+    public static List<ProviderPersonRelationship> getAssignedPatients(Provider provider)
+            throws InvalidRelationshipTypeException, PersonIsNotProviderException {
+
+        ProviderManagementService providerManagementService = Context.getService(ProviderManagementService.class);
+        PatientService patientService = Context.getService(PatientService.class);
+        List<ProviderPersonRelationship> patientsList = new ArrayList<ProviderPersonRelationship>();
+        for (RelationshipType relationshipType : provider.getProviderRole().getRelationshipTypes() ) {
+            if (!relationshipType.isRetired()) {
+                for (Relationship relationship : providerManagementService.getPatientRelationshipsForProvider(provider.getPerson(), relationshipType, null)) {
```

Comment:
No, because I need information about the relationship instance that exist between provider and patient. The getPatientsOfProvider only returns a list of patients.